src/vivarium/framework/randomness/stream.py

```diff
@@ -389,24 +411,31 @@ def _choice(
         more than one reference to `RESIDUAL_CHOICE`.
     """
     # Convert p to normalized probabilities broadcasted over index.
-    p = (
+    p_norm = (
         _set_residual_probability(_normalize_shape(p, draws.index))
         if p is not None
         else np.ones((len(draws.index), len(choices)))
     )
-    p = p / p.sum(axis=1, keepdims=True)
+    p_norm = p_norm / p_norm.sum(axis=1, keepdims=True)
 
-    p_bins = np.cumsum(p, axis=1)
+    p_bins = np.cumsum(p_norm, axis=1)
     # Use the random draw to make a choice for every row in index.
     choice_index = (draws.values[np.newaxis].T > p_bins).sum(axis=1)
 
-    return pd.Series(np.array(choices)[choice_index], index=draws.index)
+    decisions: pd.Series[Any] = pd.Series(np.array(choices)[choice_index], index=draws.index)
+
+    return decisions
 
 
 def _normalize_shape(
-    p: Union[List, Tuple, np.ndarray, pd.Series],
-    index: Union[pd.Index, pd.MultiIndex],
-) -> np.ndarray:
+    p: (
+        list[float | object]
+        | tuple[float | object]
+        | npt.NDArray[np.number[npt.NBitBase] | np.object_]
+        | pd.Series[Any]
+    ),
+    index: pd.Index[int] | pd.MultiIndex,
+) -> npt.NDArray[Any]:
```

Comment:
shouldn't this be `npt.NDArray[np.number[npt.NBitBase] | np.object_]`?

Reply:
yes